src/Spaces/dss.jl

```diff
@@ -652,62 +652,282 @@ function dss_2d!(
     return data
 end
 
-function weighted_dss!(
+"""
+    weighted_dss!(data, space, ghost_buffer = nothing)
+
+Computed weighted dss of `data`.
```

Comment:
Can you expand a bit on this? Say that this function does the whole thing, but the sub functions can be used to better overlap multiple dss calls.